### PR TITLE
testing building env on macOS and windows

### PR DESCRIPTION
**Pull request by vuillaut**
Potentially fixing #16

### CODE REVIEW

**Comment by vuillaut** [comment]
Hi @eduardo-rodrigues 
Contrary to what I announced earlier, would it be possible for your course to actually split the environments?

There are two reasons for that:
-  keeping a lighter and simpler env for the core courses
- solving install issues on different OS

**Comment by eduardo-rodrigues** [comment]
> Hi @eduardo-rodrigues
> Contrary to what I announced earlier, would it be possible for your course to actually split the environments?

We can certainly do that. I'm not sure what you would suggest in practice? I could add an environment_scikit-hep.yml file for example? In any case I think I will have all the lecture courses in a repo in my Github - seems always a good idea.

> There are two reasons for that:
> 
> * keeping a lighter and simpler env for the core courses (I underestimated the number of heavy libraries for scikit-hep)

In fact scikit-hep itself *is* very light, about 10% of what you already had prior to adding it, I think ;-). On top of that we provide wheels for anything that has bindings to ensure that installations are as trivial and efficient as possible, across "all" platforms. What is much much heavier is the addition of zfit with TF and friends. (Just so that the wrong impression does not stick around.)

> * solving install issues on different OS

Ah, that could be annoying. Where did you get issues? I think zfit does not support Windows (I do have a Windows laptop).

**Comment by maxnoe** [comment]
The pyerfa issue should be fixed now, the new conda forge package with the correct numpy compilation is available.

I will do the same for gammapy now (also providing macos arm64 builds), so that should be also out today.

**Comment by vuillaut** [comment]
> > Hi @eduardo-rodrigues
> > Contrary to what I announced earlier, would it be possible for your course to actually split the environments?
> 
> We can certainly do that. I'm not sure what you would suggest in practice? I could add an environment_scikit-hep.yml file for example? In any case I think I will have all the lecture courses in a repo in my Github - seems always a good idea.

This is exactly what I am proposing in this PR, moving the bits necessary for the scikit-hep course to a subdirectory with its own environment.

> 
> > There are two reasons for that:
> > 
> > * keeping a lighter and simpler env for the core courses (I underestimated the number of heavy libraries for scikit-hep)
> 
> In fact scikit-hep itself _is_ very light, about 10% of what you already had prior to adding it, I think ;-). On top of that we provide wheels for anything that has bindings to ensure that installations are as trivial and efficient as possible, across "all" platforms. What is much much heavier is the addition of zfit with TF and friends. (Just so that the wrong impression does not stick around.)

Yes thank you for the precision, I meant for the scikit-hep course :) 

> 
> > * solving install issues on different OS
> 
> Ah, that could be annoying. Where did you get issues? I think zfit does not support Windows (I do have a Windows laptop).

I had trouble having the CI build the env for windows, in addition with issue #16 .
With this PR, the CI is working for Linux, macOS and windows for the core environment.
It's also testing the scikit-hep env for Linux and macOS - and we can work in a subsequent PR to see if we can make it work for windows.

**Comment by eduardo-rodrigues** [comment]
> With this PR, the CI is working for Linux, macOS and windows for the core environment.
It's also testing the scikit-hep env for Linux and macOS - and we can work in a subsequent PR to see if we can make it work for windows.

Sounds good, thanks. I was offline a bit but will look now ...

File: .github/workflows/conda-shep.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 5
+      matrix:
+        os: ["ubuntu-latest", "macos-latest"]
+        python-version: ["3.9"]
```
Comment (eduardo-rodrigues) [inline .github/workflows/conda-shep.yml:13]: Being maniac I would also test 3.8 as many users are bound to be on that major version.

Comment (maxnoe) [reply .github/workflows/conda-shep.yml:13]: The environment specifies 3.9, and the users are expected to setup the environment, so how can they not have 3.9?

File: scikit-hep/environment.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+name: eschool2021_shep
```
Comment (eduardo-rodrigues) [inline scikit-hep/environment.yml:1]: Shall one remove e.g. all the astro stuff as I won't be needing it?

**Review by eduardo-rodrigues — COMMENTED** [review]
Could you replace shep with skhep everywhere? That's even how we import the metapackage.

File: .github/workflows/conda-shep.yml (hunk 1 above)
Comment (eduardo-rodrigues) [reply .github/workflows/conda-shep.yml:13]: I mean, for anyone installing locally, not using Binder, will you demand that they have 3.9? If that's the case then fine.

Comment (eduardo-rodrigues) [reply .github/workflows/conda-shep.yml:13]: I always try to be as flexible as possible, hence my comment.

Comment (maxnoe) [reply .github/workflows/conda-shep.yml:13]: The users are asked to create the environment and that will install the specified python version.

Comment (eduardo-rodrigues) [reply .github/workflows/conda-shep.yml:13]: OK. I will try the setup next week anyway as on my Windows laptop I get 3.8 from Anaconda, which is what I use "daily".

File: scikit-hep/environment.yml (hunk 2 above)
Comment (vuillaut) [reply scikit-hep/environment.yml:1]: Yes sure, you may remove what is not needed there.

Comment (eduardo-rodrigues) [reply scikit-hep/environment.yml]: I will then remove unnecessary stuff when commiting a basic README, etc., so that we can proceed here. Thanks.

**Review by eduardo-rodrigues — APPROVED** [review]

**Review by eduardo-rodrigues — APPROVED** [review]
LGTM

**Pull request merged**